common/djangoapps/util/db.py

```diff
@@ -142,6 +146,29 @@ def commit_on_success(using=None, read_committed=False):
         return CommitOnSuccessManager(using, read_committed)
 
 
+@contextmanager
+def enable_outer_atomic(names):
+    """
+    Enable outer_atomics with names.
+
+    Can be used either as a decorator or a context manager.
+    See docstring of outer_atomic for details.
+
+    Arguments:
+        names (a string or list of strings): Names of outer_atomics.
+    """
+    if isinstance(names, basestring):
```

Comment:
I agree with @adampalay - calling this as below:

```
with enable_outer_atomic('abc', 'def'):
    pass
```
is nicer than:
```
with enable_outer_atomic(['abc', 'def']):
    pass
```
I wish that:
```
with enable_outer_atomic():
```
with no parameters would be an error instead of a no-op, since it essentially does nothing except clutter the code. What do you think?

And - how about `enable_named_outer_atomic` instead of `enable_outer_atomic`? I feel like it describes it better.